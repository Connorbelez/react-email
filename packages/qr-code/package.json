{
  "name": "@react-email/qr-code",
  "version": "0.0.8",
  "description": "generate qr codes",
  "sideEffects": false,
  "main": "./dist/index.js",
  "module": "./dist/index.mjs",
  "types": "./dist/index.d.ts",
  "files": [
    "dist/**"
  ],
  "exports": {
    ".": {
      "import": {
        "types": "./dist/index.d.mts",
        "default": "./dist/index.mjs"
      },
      "require": {
        "types": "./dist/index.d.ts",
        "default": "./dist/index.js"
      }
    }
  },
  "scripts": {
    "build": "tsup src/index.ts --format esm,cjs --dts --external react",
    "clean": "rm -rf dist",
    "dev": "tsup src/index.ts --format esm,cjs --dts --external react --watch",
    "lint": "eslint .",
    "test:watch": "vitest",
    "test": "vitest run"
  },
  "keywords": [
    "react",
    "email",
    "qr-code"
  ],
  "repository": {
    "type": "git",
    "url": "https://github.com/resendlabs/react-email.git",
    "directory": "packages/qr-code"
  },
  "engines": {
    "node": ">=18.0.0"
  },
  "publishConfig": {
    "access": "public"
  },
  "license": "MIT",
  "peerDependencies": {
    "react": "18.2.0"
  },
  "devDependencies": {
    "@babel/preset-react": "7.22.5",
    "@react-email/render": "0.0.10",
    "eslint-config-custom": "workspace:*",
    "tsconfig": "workspace:*",
    "typescript": "5.1.6"
  },
  "dependencies": {
    "qrcode.react": "^3.1.0"
  }
}
