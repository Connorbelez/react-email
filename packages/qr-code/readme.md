
---


<div align="center"><strong>@react-email/qr-code</strong></div>
<div align="center">A React component for rendering QR codes in emails.</div>
<br />
<div align="center">
<a href="https://react.email">Website</a> 
<span> · </span>
<!-- <a href="https://react.email/qr-code">Documentation</a>  -->
<span> · </span>
<a href="https://twitter.com/reactemail">Twitter</a>
</div>

## Install

Install the component from your command line.

#### With yarn

```sh
yarn add @react-email/qr-code -E
```

#### With npm

```sh
npm install @react-email/qr-code -E
```

## Getting Started

Add the `QrCode` component to your email template. Configure it with the desired properties.

```jsx
import { QrCode } from "@react-email/qr-code";

const Email = () => {
  return (
    <QrCode 
      value="https://example.com"
      size={200}
      correctionLevel="L"
      renderAs="svg"
      bgColor="#FFFFFF"
      fgColor="#000000"
      includeMargin={false}
    />
  );
};
```

## Features

- **Customizable Size:** Adjust the size of the QR code to fit your design.
- **Flexible Rendering:** Choose between `canvas` and `svg` rendering.
- **Styling Options:** Customize background and foreground colors.
- **Error Correction:** Set the error correction level (L, M, H).
- **Optional Margin:** Include or exclude margin around the QR code.
- **Icon Support:** Embed a custom icon in the center of the QR code.



## License

MIT License

---

Certainly! Including a detailed explanation of the props for your `QrCode` component will greatly assist users in understanding how to effectively use the component in their projects. Here’s how you can add this to your documentation:

---

## `QrCode` Component Props

The `QrCode` component accepts several props to customize its appearance and functionality:

- **`value` (required):** A string that represents the data to be encoded in the QR code. Typically a URL or text.

- **`size` (required):** A number specifying the size of the QR code in pixels. Applies to both width and height.

- **`correctionLevel`:** A string that sets the error correction capability of the QR code. Accepts one of three values: `'L'` (Low), `'M'` (Medium), or `'H'` (High). Higher levels increase the QR code's resilience to damage but also increase its complexity.

- **`renderAs`:** Specifies the rendering mode of the QR code. Can be either `'canvas'` or `'svg'`. Default is `'canvas'`.

- **`bgColor`:** Background color of the QR code. Accepts any valid CSS color string. Default is `'#FFFFFF'` (white).

- **`fgColor`:** Foreground color of the QR code (i.e., the color of the pixels). Accepts any valid CSS color string. Default is `'#000000'` (black).

- **`includeMargin`:** A boolean that determines whether to include a margin around the QR code. Default is `false`.

- **`iconSrc`:** URL of an icon to be placed in the center of the QR code. Optional.

- **`iconSize`:** Size of the icon in pixels. If not specified, it defaults to 1/10th of the `size` prop.

- **`iconBg`:** A boolean that determines whether to include a background for the icon, making parts of the QR code under the icon transparent. Default is `true`.

- **`style`:** An optional `React.CSSProperties` object to apply custom styles to the container of the QR code.

### Example Usage

```jsx
<QrCode 
  value="https://example.com"
  size={200}
  correctionLevel="M"
  renderAs="svg"
  bgColor="#FFFFFF"
  fgColor="#333333"
  includeMargin={true}
  iconSrc="https://example.com/icon.png"
  iconSize={40}
  iconBg={false}
  style={{ padding: '10px', backgroundColor: '#f0f0f0' }}
/>
```

This example creates a QR code with medium error correction level, rendered as an SVG, with custom colors, margin, an icon, and additional styling.

---

### Additional Notes:

- **Clarity and Detail:** Each prop explanation should be clear and detailed enough for users to understand its purpose and how it affects the component.
- **Defaults and Requirements:** Indicate which props are required and what the default values are for optional props.
- **Examples:** Including examples of how to use the props can be very helpful, especially for props that might be less intuitive.
