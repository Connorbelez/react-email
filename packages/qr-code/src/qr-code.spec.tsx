import { render } from "@react-email/render";
import { QrCode } from './qr-code'; // Adjust the import path as needed

test('renders without crashing', () => {
    render(<QrCode value="test" size={300} correctionLevel="L" />);
});
describe('<QrCode> component', () => {
    // Test cases will go here
    it('renders QR code with required props', () => {
        const actualOutput = render(
            <QrCode value="test" size={300} correctionLevel="L" />
        );// Check if the value is rendered
        expect(actualOutput).toContain('canvas');
        expect(actualOutput).toContain('width="300"');
        expect(actualOutput).toContain('height="300"');
        expect(actualOutput).toMatchSnapshot();
    });
    //
    it('renders QR code with optional props', () => {
        const actualOutput = render(
            <QrCode
                value="test"
                size={300}
                correctionLevel="L"
                renderAs="canvas"
                bgColor="#FFFFFF"
                fgColor="#000000"
                style={{ backgroundColor: 'blue' }}
            />
        );
        expect(actualOutput).toContain('canvas');
        expect(actualOutput).toContain('width="300"');
        expect(actualOutput).toContain('height="300"');
        expect(actualOutput).toMatchSnapshot();
        expect(actualOutput).toContain('style="background-color:blue"');

    });


});
