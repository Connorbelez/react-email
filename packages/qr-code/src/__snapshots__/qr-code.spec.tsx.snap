// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<QrCode> component > renders QR code with optional props 1`] = `"<!DOCTYPE html PUBLIC \\"-//W3C//DTD XHTML 1.0 Transitional//EN\\" \\"http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd\\"><div style=\\"background-color:blue\\"><canvas style=\\"height:300px;width:300px\\" height=\\"300\\" width=\\"300\\"></canvas></div>"`;

exports[`<QrCode> component > renders QR code with required props 1`] = `"<!DOCTYPE html PUBLIC \\"-//W3C//DTD XHTML 1.0 Transitional//EN\\" \\"http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd\\"><div><canvas style=\\"height:300px;width:300px\\" height=\\"300\\" width=\\"300\\"></canvas></div>"`;
